checkout-core/src/main/java/com/adyen/checkout/core/api/OkHttpClient.kt

```diff
@@ -49,7 +49,14 @@ internal class OkHttpClient(
                 ?.bytes()
                 ?: ByteArray(0)
         } else {
-            throw IOException(response.body?.string())
+            val errorBody = try {
+                response.body?.string()
+                    ?.let { JSONObject(it) }
+                    ?.let { ErrorResponseBody.SERIALIZER.deserialize(it) }
+            } catch (e: Throwable) {
```

Comment:
Why `Throwable`? Shouldn't it be `IOException`?

Reply:
There are 3 different exceptions that should be handled in the same way, so I thought this would be okay

Reply:
Can we do something like this in Kotlin? `catch (ExceptionType1 | Exceptiontype2 ex)`

Reply:
Nope each exception needs it's own catch block :(
```
try { }
catch(e: SomeException) { }
catch(e: SomeOtherException) { }
```